### DISCUSSION TITLE
E_BAD_CSRF_TOKEN: Invalid CSRF Token

### DISCUSSION THREAD

**Discussion by feniljariwala82** [adonisjs/core#3162] [Help]
**Purpose**: Make POST, PUT, DELETE request from React App  **with XSRF protection**

**Description**: I am making a request from the React client to the Adonis server. Also, I followed the guide about adding **XSRF-TOKEN** as an **"X-XSRF-TOKEN"** header. But it's still throwing an error of **E_BAD_CSRF_TOKEN: Invalid CSRF Token**.

**Note**:
enableXSRFcookie: enabled,

 I tried it in two ways,
 1) I copy-pasted the **XSRF-TOKEN** cookie value and added it as a header. 
2) I added it as a decoded URL. 

But it is throwing the same error **E_BAD_CSRF_TOKEN**.

Can anyone guide me here? Your help will be appreciated.

**Comment by thetutlage** [comment]
Can you share the dev-tools screenshot of the request that react makes? Make sure the screenshot shows all the request headers

**Reply by feniljariwala82** [reply]
![pic1](https://user-images.githubusercontent.com/42708686/143081095-96e67e85-e920-4c48-835a-76e6785910eb.PNG)
![pic2](https://user-images.githubusercontent.com/42708686/143081104-af564365-cdf5-4b84-b2e6-ed6e51c5774e.PNG)
![pic3](https://user-images.githubusercontent.com/42708686/143081109-9acda113-1477-498d-9c78-95f9689777fa.PNG)

I am having this error in Ajax request while uploading image. Can anybody guide me here?
